File: server/order-service/src/main/java/com/sudo248/orderservice/controller/order/dto/OrderSupplierDto.java
package com.sudo248.orderservice.controller.order.dto;

import com.sudo248.orderservice.repository.entity.order.OrderStatus;
import com.sudo248.orderservice.repository.entity.order.Shipment;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
// for customer
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class OrderSupplierDto {
    private String orderSupplierId;
    private SupplierInfoDto supplier;
    private PromotionDto promotion;
    private double totalPrice;
    private Double revenue;
    private Shipment shipment;
    private LocalDateTime expectedReceiveDateTime;
    private OrderStatus status;
    private List<OrderCartProductDto> orderCartProducts;
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/data/dto/order/OrderSupplierDto.kt
package com.sudo248.sudoo.data.dto.order

import com.sudo248.sudoo.data.dto.discovery.SupplierInfoDto
import com.sudo248.sudoo.data.dto.promotion.PromotionDto
import com.sudo248.sudoo.domain.entity.order.Shipment
import java.time.LocalDateTime

data class OrderSupplierDto(
    val orderSupplierId: String,
    val supplier: SupplierInfoDto,
    val promotion: PromotionDto? = null,
    val shipment: Shipment,
    val totalPrice: Double,
    val expectedReceiveDateTime: LocalDateTime,
    val orderCartProducts: List<OrderCartProductDto>
)
